CMakeLists.txt

```diff
@@ -31,7 +31,15 @@ set(SPIRV_TOOLS "SPIRV-Tools")
 include(GNUInstallDirs)
 
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-set(CMAKE_CXX_STANDARD 11)
+
+# Require at least C++11
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 11)
+endif()
+if(${CMAKE_CXX_STANDARD} LESS 11)
+  message(FATAL_ERROR "SPIRV-Tools requires C++11 or later, but is configured for C++${CMAKE_CXX_STANDARD})")
+endif()
```

Comment:
All targets should just have `target_compile_features(<target> PUBLIC cxx_std_11)`, CMakeLists of libraries should never set CMAKE_CXX_STANDARD.

Reply:
Question:  Should CMAKE_CXX_STANDARD be set when building executables?
E.g. the command line tools are built via https://github.com/KhronosGroup/SPIRV-Tools/blob/main/tools/CMakeLists.txt

Reply:
I'm confused.  Why does CMAKE_CXX_STANDARD exist, then?  Or is it legacy?

Ah, target_compile_features is new in CMake 3.8. 
https://cmake.org/cmake/help/latest/release/3.8.html?highlight=cxx_std_11#c-c